### PR TITLE
Improvements to libass codepath

### PR DESCRIPTION
**Pull request by adipose**
1. Doesn't rely on internal sts segments
2. Fix for SRT subs aspect ratio (set to srtresx/y before--still not sure if we can get rid of this, but at least avoiding setting storage size fixes aspect ratio changes with SRT)
3. Missing commit for resetting libass when initializing RTS

### CODE REVIEW

File: src/Subtitles/RTS.cpp [hunk 1]
```diff
@@ -1872,6 +1872,9 @@ bool CRenderedTextSubtitle::Init(CSize size, const CRect& vidrect)
         Deinit();
         m_size = newSize;
         m_vidrect = newVidRect;
+#if USE_LIBASS
+        m_SSAUtil.ResetASS();
```
Comment (clsid2) [inline src/Subtitles/RTS.cpp]: Rename `SSAUtil` class to something like `LibassContext`? Then it is clearer what it is for.
Same for `m_SSAUtil`.

File: src/Subtitles/RTS.cpp [hunk 2]
```diff
@@ -3110,6 +3113,14 @@ STDMETHODIMP CRenderedTextSubtitle::NonDelegatingQueryInterface(REFIID riid, voi
 
 STDMETHODIMP_(POSITION) CRenderedTextSubtitle::GetStartPosition(REFERENCE_TIME rt, double fps)
 {
+#if USE_LIBASS
+    POSITION p;
```
Comment (clsid2) [inline src/Subtitles/RTS.cpp]:
```
if (m_SSAUtil.IsLibassActive()) {
     // do libass stuff
    return something;
}
// do ISR stuff
```

File: src/Subtitles/SubtitleInputPin.cpp [hunk 3]
```diff
@@ -480,44 +480,42 @@ REFERENCE_TIME CSubtitleInputPin::DecodeSample(const std::unique_ptr<SubtitleSam
...
-                    stse.readorder = wcstol(sl.RemoveHead(), nullptr, 10);
-                    stse.layer = wcstol(sl.RemoveHead(), nullptr, 10);
-                    stse.style = sl.RemoveHead(); // no trim, its value is a lookup key
-                    stse.actor = sl.RemoveHead().Trim();
-                    stse.marginRect.left = wcstol(sl.RemoveHead(), nullptr, 10);
-                    stse.marginRect.right = wcstol(sl.RemoveHead(), nullptr, 10);
-                    stse.marginRect.top = stse.marginRect.bottom = wcstol(sl.RemoveHead(), nullptr, 10);
-                    if (fields == 10) {
-                        stse.marginRect.bottom = wcstol(sl.RemoveHead(), nullptr, 10);
+#if USE_LIBASS
```
Comment (clsid2) [inline src/Subtitles/SubtitleInputPin.cpp:484]: similar change is needed a few lines above for MEDIASUBTYPE_UTF8

**Comment by clsid2** [comment]
ASS subs still do not appear until a seek has been done. I have not investigated yet what is the cause.

Default style for SRT uses bold font. With libass the font weight seems to be ignored? Same for italic.

**Comment by adipose** [comment]
> ASS subs still do not appear until a seek has been done. I have not investigated yet what is the cause.
> 
> Default style for SRT uses bold font. With libass the font weight seems to be ignored? Same for italic.

I'll look into it.  For SRT, do you mean MPC-HC default is bold, or that somehow libass is using bold?  I did notice a "regression" on subs recently that they seemed bolded when they didn't use to be...I can't figure out if something changed or not.

**Comment by clsid2** [comment]
With libass the font is shown as regular, even if default style in the player is set to bold and/or italic.

Difference is easy to see by turning off libass option and reloading the video.

Maybe just something missing in the SRT to ASS conversion. The other style options do all have effect correctly.

**Comment by adipose** [comment]
> ASS subs still do not appear until a seek has been done. I have not investigated yet what is the cause.

I couldn't reproduce this on a short video.  Do you have an example?

**Comment by clsid2** [comment]
[Sample](https://www.sendspace.com/file/wvbjv6)
First sub at 0:02 is skipped if played from start.

Also happens with other files with embedded subs. It seems it is just the first (few) sub lines that are skipped. Interestingly, if I let it play until first one shows up, then reload with remember position, then again a few are skipped.
[Another sample](https://www.sendspace.com/file/hb9myx)

**Comment by adipose** [comment]
> [Sample](https://www.sendspace.com/file/wvbjv6) First sub at 0:02 is skipped if played from start.
> 
> Also happens with other files with embedded subs. It seems it is just the first (few) sub lines that are skipped. Interestingly, if I let it play until first one shows up, then reload with remember position, then again a few are skipped. [Another sample](https://www.sendspace.com/file/hb9myx)

Ok, I see it now.

**Comment by adipose** [comment]
> [Sample](https://www.sendspace.com/file/wvbjv6) First sub at 0:02 is skipped if played from start.
> 
> Also happens with other files with embedded subs. It seems it is just the first (few) sub lines that are skipped. Interestingly, if I let it play until first one shows up, then reload with remember position, then again a few are skipped. [Another sample](https://www.sendspace.com/file/hb9myx)

https://github.com/clsid2/mpc-hc/pull/2189

I think this should fix it.  The reset after a style change needs to reload already processed samples.

**Pull request merged**